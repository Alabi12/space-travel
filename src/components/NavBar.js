import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from './planet.png';

const NavBar = () => (
  <nav className="navBar">
    <a href="/">
      <img src={logo} className="App-logo" alt="logo" />
    </a>
    <h1>Space Travelers` Hub</h1>
    <ul className="navMenu">
      <li>
        <NavLink
          to="/rockets"
          className="navItem"
          activeclassname="active"
        >
          Rockets
        </NavLink>
      </li>
      <li>
        <NavLink to="/missions" className="navItem" activeclassname="active">
          Missions
        </NavLink>
      </li>
      <li>
        <NavLink to="/myprofile" className="navItem navItemMyProfile" activeclassname="active">
          My Profile
        </NavLink>
      </li>
    </ul>
  </nav>
);

export default NavBar;
