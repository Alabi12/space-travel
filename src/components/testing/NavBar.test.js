import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import NavBar from '../NavBar';

describe('Testing NavBar component', () => {
  test('Snapshot testing', () => {
    const myRenderer = renderer.create(
      <Router>
        <NavBar />
      </Router>,
    );
    const testComponent = myRenderer.toJSON();
    expect(testComponent).toMatchSnapshot();
  });
});
