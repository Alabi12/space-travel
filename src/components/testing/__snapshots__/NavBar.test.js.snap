// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NavBar component Snapshot testing 1`] = `
<nav
  className="navBar"
>
  <a
    href="/"
  >
    <img
      alt="logo"
      className="App-logo"
      src="planet.png"
    />
  </a>
  <h1>
    Space Travelers\` Hub
  </h1>
  <ul
    className="navMenu"
  >
    <li>
      <a
        activeclassname="active"
        className="navItem"
        href="/rockets"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        activeclassname="active"
        className="navItem"
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <li>
      <a
        activeclassname="active"
        className="navItem navItemMyProfile"
        href="/myprofile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </li>
  </ul>
</nav>
`;

exports[`test of nav bar component compare with snapshot 1 1`] = `
<nav
  className="navBar"
>
  <a
    href="/"
  >
    <img
      alt="logo"
      className="App-logo"
      src="planet.png"
    />
  </a>
  <h1>
    Space Travelers\` Hub
  </h1>
  <ul
    className="navMenu"
  >
    <li>
      <a
        activeclassname="active"
        className="navItem"
        href="/rockets"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        activeclassname="active"
        className="navItem"
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <li>
      <a
        activeclassname="active"
        className="navItem navItemMyProfile"
        href="/myprofile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </li>
  </ul>
</nav>
`;
